### DISCUSSION TITLE
Composable useItems does not accept parameters

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#8860] [Archive]
when using ```useItems(ref,filter)``` does not except any parameters 

```js

import {useitems} from "'@directus/extensions-sdk'"
import {ref} from "vue"

....


setup(){
const collection = ref('someCollection')
const filter = {
filter:{
id:{
_eq: 1
}}}

const {items} = useItems(collection,filter)
}
```

items will always be empty
putting a ``` console.log(collection)``` || ```console.log(query)``` inside ```use-items.ts``` logs a empty line
it seems the parameters are not passed.

### Errors Shown
```
TypeError: m.value is undefined
    http://localhost:6000/admin/assets/use-sync.6d51d5e5.js:1
```
m.value is not defined because parameter is not passed
when a normal string is send instead of a ref no error is shown.
still logging a empty line

**Comment by rijkvanzanten** [comment]
I'm having a bit of trouble following your exact example, but make sure to pass the query object as the second parameters, not just a flag object of a filter structure, see https://github.com/directus/directus/blob/567ce009222cf69c80b3334498bebc3b89c979b3/app/src/layouts/tabular/index.ts#L47-L57 for example. The second thing to note is that use-items expects the query object to contain reactive properties: https://github.com/directus/directus/blob/567ce009222cf69c80b3334498bebc3b89c979b3/packages/shared/src/composables/use-items.ts#L26-L33

**Reply by cpt-n3mo** [reply]
Good day!

I think im following what your saying..
the filter object needs to be reactive
and second parameter of useItems need to be a object
```
const collection = ref("someCollection");
const myFilter = ref({
        id: {
          _eq: 1,
        },
      },
    );
const { items } = useItems(collection, {
      filter: myFilter,
    });

```
but this still gives me ``` TypeError: m is undefined ```

kind regards..

**Reply by rijkvanzanten** [reply]
The other query params are also required, so limit, sort, etc

**Reply by cpt-n3mo** [reply]
Mmm 
ok this is somewhat strange..
i imported ``` node_modules/@directus/shared/dist/esm/composables/use-items.js ````
instead the one from extentions-sdk

and used the following query params
```
const fields = ref(['*.*'])
const limit = ref() 
const sort = ref([]) 
const search = ref('') 
const page = ref('') 
const filter = ref({
        id: {
          _eq: 1,
        },
      },
    );
```

this fetches the data with now problem..
doing the same with the one from extentions-sdk makes it fail.
this is highlighted in the debugger
```
1===m.value.includes("*")&&
```
is the ts version giving more constraints to the query params ..?

kind regards..

**Reply by cpt-n3mo** [reply]
looking at the 2 files there seems to be some work done in this area. 

extentions-sdk
```
let fieldsToFetch = [...(unref(fields) ?? [])];

		// Make sure the primary key is always fetched
		if (
			!unref(fields)?.includes('*') &&
			primaryKeyField.value &&
			fieldsToFetch.includes(primaryKeyField.value.field) === false
		) {
			fieldsToFetch.push(primaryKeyField.value.field);
		}
```
from directus files
```
let fieldsToFetch = [...((_a = unref(fields)) !== null && _a !== void 0 ? _a : [])];
        // Make sure the primary key is always fetched
        if (!((_b = unref(fields)) === null || _b === void 0 ? void 0 : _b.includes('*')) &&
            primaryKeyField.value &&
            fieldsToFetch.includes(primaryKeyField.value.field) === false) {
            fieldsToFetch.push(primaryKeyField.value.field);
        }
```
i suspect the debugger output from above post is this section 
```
!unref(fields)?.includes('*') &&
			primaryKeyField.value
```

kind regards..

**Reply by cpt-n3mo** [reply]
ok...

bit more searching, and i noticed that my custom module had ```  "@directus/extensions-sdk": "^9.0.0-rc.98"  ``` removing it solved the problem.   ``` "directus": "^9.0.0-rc.95" ``` and ```  "@directus/extensions-sdk": "^9.0.0-rc.98"  ```  where in conflict apparently.
now using it from the main app and works like a charm!
Thank you for your time! again.. sorry for that..  

on the subject of sharing some gained understanding that is not yet documented, as i was reading that you guys would like ppl to contribute in this as wel, is there a place to do so .. ?

kind regards!

**Reply by benhaynes** [reply]
Hey @cpt-n3mo — we're overhauling the _structure_ (and some content) of the docs now, which should be released this week. We would LOVE any help adding missing context to the docs if you have time. They are in markdown files within the Directus codebase itself.

**Reply by cpt-n3mo** [reply]
ah Perfect!
thats very good news..!
I will deff try and do that then..  :)

kind regards !
